src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -3786,92 +3786,59 @@ internal static int LastIndexOfAnyExceptInRangeUnsignedNumber<T>(ref T searchSpa
                 {
                     Vector512<T> targetVector = Vector512.Create(value);
                     ref T oneVectorAwayFromEnd = ref Unsafe.Subtract(ref end, Vector512<T>.Count);
-                    do
+                    while (Unsafe.IsAddressLessThan(ref current, ref oneVectorAwayFromEnd))
```

Comment:
Why the change from a do...while to a while? Does it help? We just validated there's enough room, so we shouldn't need the starting check, right?

Reply:
Correctness-wise, it doesn't matter, the overlapped block can handle any pattern.

Theoretically the `while (less than)`, we can skip the main loop iteration if the input is exactly one `VectorXYZ<T>` long. But it does mean an extra check for all the other cases. Perf-wise it makes practically no difference in this case.

The `do..while(less than)` saves a dozen bytes of assembly for the removed branches, so I went with that.